src/Plugin/Validators/ValidDataFile.php

```diff
@@ -195,10 +274,161 @@ public function validateFile(int|null $fid) {
 
     // Validator response values if data file is valid.
     return [
-      'case' => 'Data file is valid',
+      'case' => self::$mapping['case-valid']['dev-case'],
       'valid' => TRUE,
       'failedItems' => [],
     ];
   }
 
+  /**
+   * Processes failed validation from ValidDataFile into a render array.
+   *
+   * @param array $validation_status
+   *   An associative array that stores the validation failures by the
+   *   ValidDataFile validator. It is an associative array returned by the
+   *   validator. Here is the overall structure:
+   *     - 'case': a developer-focused string describing the case checked.
+   *     - 'valid': FALSE to indicate that validation failed.
+   *     - 'failedItems': an array of items that failed with one or more of the
+   *       following keys:
+   *         - 'filename': The provided name of the file.
+   *         - 'fid': The fid of the provided file.
+   *         - 'mime': The mime type of the input file if it is not supported.
+   *         - 'extension': The extension of the input file if not supported.
+   *         @see validateFile()
+   * @param array $tokens
+   *   [OPTIONAL] An array of values to use for token replacement.
+   *   @see ValidDataFile::$mapping
+   *   The following token keys will substitute the entire existing case message
+   *   to the user with the value of that token.
+   *     - 'case-invalid-fid': the message when a file id is not a valid value.
+   *     - 'case-failed-fid': the message when a file id failed to load a
+   *       file object.
+   *     - 'case-empty-file': the message when a file has no data or is an
+   *       empty file.
+   *     - 'case-unsupported-mime': the message when file MIME type is
+   *       not supported.
+   *     - 'case-unsupported-extension': the message when mime type and
+   *       extension are not supported.
+   *     - 'case-locked-file': the message when a file could not be opened.
+   *     - 'contact-admin': the phrase to use when the user needs a
+   *       priviledged administrator to fix the problem.
```

Comment:
Since this isn't a case message token, this needs a separate sentence to introduce it. Here is what was used in GermplasmNameExists:
```suggestion
   *     The following tokens can be specfied as keys, with value as the
   *     replacement value for the token.
   *     - 'contact-admin': the phrase to use when the user needs a
   *       priviledged administrator to fix the problem.
```